src/models/json-api.model.ts

```diff
@@ -3,17 +3,35 @@ import 'reflect-metadata';
 import { Headers } from '@angular/http';
 import { Observable } from 'rxjs/Observable';
 import { JsonApiDatastore, ModelType } from '../services/json-api-datastore.service';
+import { DocumentModel } from './document.model';
+import { LinksModel } from './links.model';
+import { LinkModel } from './link.model';
 
 export class JsonApiModel {
 
   id: string;
+  _links: LinksModel = new LinksModel;
+  _document: DocumentModel<JsonApiModel>;
   [key: string]: any;
 
-  constructor(private _datastore: JsonApiDatastore, data?: any) {
+  constructor(private _datastore: JsonApiDatastore, data?: any, document: DocumentModel<JsonApiModel> = null) {
```

Comment:
Moreover, instead of `document: DocumentModel<JsonApiModel> = null` you can write: `document?: DocumentModel<JsonApiModel>`